domehunter/__init__.py

```diff
@@ -86,70 +230,291 @@ def is_home(self):
 
     """These map directly onto the AbstractMethods created by RPC."""
 
     def abort(self):
-        """Stop everything."""
+        """Stop everything by switching the dome motor on/off relay to off."""
+        # note, might want another way to do this incase the relay fails/sticks
+        # one way might be cut power to the automationHAT so the motor relays
+        # will receive no voltage even if the relay is in the open position?
+        self._stop_moving()
         pass
 
-    def getAzEl(self):
-        """Return AZ and Elevation."""
+    def getAz(self):
+        """Return current Azimuth of the Dome."""
+        if self.dome_az is None:
+            self.calibrate()
+        return self.dome_az
+
+    def GotoAz(self, az):
+        "Send Dome to a requested Azimuth position."
+        if self.dome_az is None:
+            self.calibrate()
+        delta_az = az - self.dome_az
+
+        # determine whether CW or CCW gives the short path to desired az
+        if abs(delta_az) > 180:
+            if delta_az > 0:
+                delta_az -= 360
+            else:
+                delta_az += 360
+
+        # if updated delta_az is positive, direction is CW
+        if delta_az > 0:
+            # converted delta_az to equivilant in encoder ticks
+            ticks = self._az_to_ticks(delta_az)
+            target_position = self.encoder_count + ticks
+            self._move_cw()
+            # wait until encoder count matches desired delta az
+            while self.encoder_count < target_position:
+                if self.testing:
+                    # if testing simulate a tick for every cycle of while loop
+                    self._simulate_ticks(num_ticks=1)
+                pass
+            self._stop_moving()
+            # compare original count to current just in case we got more ticks
+            # than we asked for
+            old_encoder_count = target_position - ticks
+            # update dome_az based on the actual number of ticks counted
+            self.dome_az += self._ticks_to_az(
+                self.encoder_count - old_encoder_count)
+            # take mod360 of dome_az to keep 0 <= dome_az < 360
+            self.dome_az %= 360
+            # update encoder_count to match the dome_az
+            self.encoder_count = self._az_to_ticks(self.dome_az)
+            pass
+
+        # if updated delta_az is negative, direction is CCW
+        if delta_az < 0:
+            # converted delta_az to equivilant in encoder ticks
+            ticks = self._az_to_ticks(delta_az)
+            target_position = self.encoder_count + ticks
+            self._move_ccw()
+            # wait until encoder count matches desired delta az
+            while self.encoder_count >= target_position:
+                if self.testing:
+                    # if testing simulate a tick for every cycle of while loop
+                    self._simulate_ticks(num_ticks=1)
+                pass
```

Comment:
```suggestion
                else:
                       time.sleep(0.1)
```

A micro break, otherwise CPUs are burning on a simple while loop.

Reply:
Need to build-in a timeout error here eventually: #23